test/unit/inbound/process-submitted-settlement.test.js

```diff
@@ -0,0 +1,137 @@
+  })
+
+  test('should log "Duplicate settlement received, skipping {duplicateSettlement.reference}" when getExistingSettlement returns a settlement ', async () => {
+    const consoleSpy = jest.spyOn(console, 'info')
+    const duplicateSettlement = settlement
+    getExistingSettlement.mockResolvedValue(duplicateSettlement)
+
+    await processSubmittedSettlement(settlement)
+
+    expect(consoleSpy).toHaveBeenCalledWith(`Duplicate settlement received, skipping ${duplicateSettlement.reference}`)
+  })
+
+  test('should throw when duplicate settlement found ', async () => {
+    getExistingSettlement.mockRejectedValue(new Error('Duplicate settlement recieved'))
+    const wrapper = async () => {
+      await processSubmittedSettlement(settlement)
+    }
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw an error when duplicate settlement found ', async () => {
+    getExistingSettlement.mockRejectedValue(new Error('Duplicate settlement recieved'))
+    const wrapper = async () => {
+      await processSubmittedSettlement(settlement)
+    }
+    expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should callmockTransaction.rollback when a duplicate settlement is found ', async () => {
+    const duplicateSettlement = settlement
+    getExistingSettlement.mockResolvedValue(duplicateSettlement)
+    await processSubmittedSettlement(settlement)
+    expect(mockTransaction.rollback).toHaveBeenCalled()
+  })
+
+  test('should callmockTransaction.rollback once when a duplicate settlement is found ', async () => {
+    const duplicateSettlement = settlement
+    getExistingSettlement.mockResolvedValue(duplicateSettlement)
+    await processSubmittedSettlement(settlement)
+    expect(mockTransaction.rollback).toHaveBeenCalledTimes(1)
+  })
+
+  test('should callmockTransaction.commit when no existing settlement is found ', async () => {
+    await processSubmittedSettlement(settlement)
+    expect(mockTransaction.commit).toHaveBeenCalled()
+  })
+
+  test('should call mockTransaction.commit once when no existing settlement is found ', async () => {
+    await processSubmittedSettlement(settlement)
+    expect(mockTransaction.commit).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call saveSettlement when a valid settlement is given ', async () => {
+    await processSubmittedSettlement(settlement)
+    expect(saveSettlement).toHaveBeenCalled()
+  })
+
+  test('should call saveSettlement once when a valid settlement is given ', async () => {
+    await processSubmittedSettlement(settlement)
+    expect(saveSettlement).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call saveSettlement with settlement and mocktransaction ', async () => {
+    await processSubmittedSettlement(settlement)
+    expect(saveSettlement).toHaveBeenCalledWith(settlement, mockTransaction)
+  })
+
+  test('should throw when saveSettlement is called with invalid settlement ', async () => {
```

Comment:
I think this test name could be a little less specific and just mention when `saveSettlement` throws becuz remember: we've mocked this, we don't actually know (if only looking at this test) what it does and therefore why it would throw

"should throw when saveSettlement throws"

then need tests for:

- "should throw Error when saveSettlement throws"
- "should throw error "Database save down issue" when saveSettlement throws error with "Database save down issue"'